server/test/controllers/admin/AdminImportControllerTest.java

```diff
@@ -155,41 +160,859 @@ public void hxImportProgram_notEnoughInfoToCreateProgramDef_error() {
+
+    ProgramDefinition programDefinition =
+        database
+            .find(ProgramModel.class)
+            .where()
+            .eq("name", "visibility-eligibility")
+            .findOne()
+            .getProgramDefinition();
+    Long eligibilityQuestionId =
+        programDefinition
+            .getBlockDefinition(1)
+            .eligibilityDefinition()
+            .get()
+            .predicate()
+            .rootNode()
+            .getLeafOperationNode()
+            .questionId();
+    Long visibilityQuestionId =
+        programDefinition
+            .getBlockDefinition(2)
+            .visibilityPredicate()
+            .get()
+            .rootNode()
+            .getLeafOperationNode()
+            .questionId();
+    Long savedQuestionId =
+        database
+            .find(QuestionModel.class)
+            .where()
+            .eq("name", "id-test")
+            .findOne()
+            .getQuestionDefinition()
+            .getId();
+
+    assertThat(eligibilityQuestionId).isEqualTo(savedQuestionId);
+    assertThat(visibilityQuestionId).isEqualTo(savedQuestionId);
   }
+
+  @Test
+  public void saveProgram_discardsPaiTagsOnImportedQuestions() {
+    when(mockSettingsManifest.getProgramMigrationEnabled(any())).thenReturn(true);
+
+    Result result =
+        controller.saveProgram(
+            addCSRFToken(
+                    fakeRequest()
+                        .method("POST")
+                        .bodyForm(ImmutableMap.of("programJson", PROGRAM_JSON_WITH_PAI_TAGS)))
+                .build());
+
+    assertThat(result.status()).isEqualTo(OK);
+
+    QuestionDefinition questionDefinition =
+        database
+            .find(QuestionModel.class)
+            .where()
+            .eq("name", "dob")
+            .findOne()
+            .getQuestionDefinition();
+
+    assertThat(questionDefinition.getPrimaryApplicantInfoTags()).isEqualTo(ImmutableSet.of());
+  }
+
+  @Test
+  public void saveProgram_preservesUniversalSettingOnImportedQuestions() {
+    when(mockSettingsManifest.getProgramMigrationEnabled(any())).thenReturn(true);
+
+    Result result =
+        controller.saveProgram(
+            addCSRFToken(
+                    fakeRequest()
+                        .method("POST")
+                        // Questions must be marked as "universal" before being tagged with a PAI
+                        // tag, so we can reuse the PROGRAM_JSON_WITH_PAI_TAGS json
+                        .bodyForm(ImmutableMap.of("programJson", PROGRAM_JSON_WITH_PAI_TAGS)))
+                .build());
+
+    assertThat(result.status()).isEqualTo(OK);
+
+    QuestionDefinition questionDefinition =
+        database
+            .find(QuestionModel.class)
+            .where()
+            .eq("name", "dob")
+            .findOne()
+            .getQuestionDefinition();
+
+    assertThat(questionDefinition.isUniversal()).isTrue();
+  }
+
+  public static final String PROGRAM_JSON_WITHOUT_QUESTIONS =
+      "{\n"
```

Comment:
maybe as a followup PR, but it'd be nice to convert this to java text blocks like the `JsonExporterServiceTest` uses now.

Reply:
I'll add that as a todo in a follow-up issue.